test/builtin_dataset_mocks.py

```diff
@@ -1000,7 +996,7 @@ def dtd(info, root, _):
 def fer2013(info, root, config):
     num_samples = 5 if config.split == "train" else 3
 
-    path = root / f"{config.split}.txt"
+    path = root / f"{config.split}.csv"
```

Comment:
This change and everything below in this file are actual bugs in our mock data generation that were hidden by our custom loading logic.

Reply:
Do you know why the tests were passing before despite the files had the wrong name?

Reply:
Before the resources where not collected by the regular logic, but rather "hand-fed" to `_make_datapipe`. So as long as the data in `mock_resources[0]` corresponded to `Dataset.resources(...)[0]` the test suite didn't notice.